services/softlayer_virtual_guest.go

```diff
@@ -380,7 +380,7 @@ func (slvgs *softLayer_Virtual_Guest_Service) AttachEphemeralDisk(instanceId int
 		return err
 	}
 
-	order := datatypes.SoftLayer_Product_Order{
+	order := datatypes.SoftLayer_Ephemeral_Disk_Order{
```

Comment:
Since you are making this change to the `AttachEphemeralDisk` I expected it's test to be also changed. Correct? Otherwise, the test is not sufficient since it should not be passing with this additional call to `PlaceEphemeralDiskOrder `